README.md
# round_path_widget

A new Flutter package.

![screen_shot.png](https://upload-images.jianshu.io/upload_images/5459987-b8b6d6c5efd846c0.png?imageMogr2/auto-orient/strip%7CimageView2/2/w/1240)

## Getting Started

This project is a starting point for a Dart
[package](https://flutter.dev/developing-packages/),
a library module containing code that can be shared easily across
multiple Flutter or Dart projects.

For help getting started with Flutter, view our 
[online documentation](https://flutter.dev/docs), which offers tutorials, 
samples, guidance on mobile development, and a full API reference.
### 说明
此package支持圆形展示设置，圆角矩阵展示设置以及多种不同圆角展示设置，分别对应三个枚举类PathShapeEnum.CirclePath,PathShapeEnum.RoundRect,PathShapeEnum.PartRoundRect

### 使用方式
```
       #圆形设置
       RoundPathWidget(
          child: Image.asset(
          'assets/images/girl.jpeg',
          fit: BoxFit.cover,
          width: 60,
          height: 60,
        ),
      ),
      #矩形圆角设置：定义类型以及radius
      RoundPathWidget(
        pathShape: PathShapeEnum.RoundRect,
        radius: 10,
        child: Image.asset(
          'assets/images/girl.jpeg',
          fit: BoxFit.cover,
          width: 70,
          height: 70,
        ),
      ),
      #定义多种圆角类型：leftTopRadius，rightTopRadius，leftBottomRadius，rightBottomRadius
      RoundPathWidget(
        pathShape: PathShapeEnum.PartRoundRect,
        leftTopRadius: 20,
        rightBottomRadius: 20,
        child: Image.asset(
          'assets/images/girl.jpeg',
          fit: BoxFit.cover,
          width: 110,
          height: 110,
        ),
      ),
      RoundPathWidget(
        pathShape: PathShapeEnum.PartRoundRect,
        leftTopRadius: 20,
        leftBottomRadius: 20,
        child: Image.asset(
          'assets/images/girl.jpeg',
          fit: BoxFit.cover,
          width: 90,
          height: 90,
        ),
      ),
      RoundPathWidget(
        pathShape: PathShapeEnum.PartRoundRect,
        leftTopRadius: 20,
        leftBottomRadius: 10,
        rightBottomRadius: 10,
        child: Image.asset(
          'assets/images/girl.jpeg',
          fit: BoxFit.cover,
          width: 150,
          height: 150,
        ),
      ),
      
      # 对布局边缘惊醒设置
         RoundPathWidget(
                      pathShape: PathShapeEnum.PartRoundRect,
                      rightTopRadius: 15,
                      rightBottomRadius: 15,
                      child: Container(
                        width: 300,
                        padding: const EdgeInsets.all(5),
                        color: Colors.blue,
                        child: Row(
                          children: <Widget>[
                            Expanded(
                                child: Column(
                              children: <Widget>[
                                Center(
                                  child: Text("测试多种布局"),
                                ),
                                Text('支持不同角，不同圆弧大小')
                              ],
                            )),
                            Image.asset(
                              'assets/images/girl.jpeg',
                              fit: BoxFit.cover,
                              width: 60,
                              height: 60,
                            )
                          ],
                        ),
                      ),
                    ),
      
```


CHANGELOG.md
## [0.0.1] - TODO: Add release date.

* TODO: Describe initial release.
## [0.0.3]
* add description
## [0.1.0]
* add example 

test/round_path_widget_test.dart
import 'package:flutter/material.dart';
import 'package:flutter/widgets.dart';
import 'package:flutter_test/flutter_test.dart';

import 'package:round_path_widget/round_path_widget.dart';

void main() {
//  runApp(TestApp());
}

class TestApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('支持各种圆角布局样式Widget'),
      ),
      body: SingleChildScrollView(
        child: Center(
          child: Column(
            children: <Widget>[
              RoundPathWidget(
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 60,
                  height: 60,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.RoundRect,
                radius: 10,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 70,
                  height: 70,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                rightBottomRadius: 20,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 110,
                  height: 110,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                leftBottomRadius: 20,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 90,
                  height: 90,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                leftBottomRadius: 10,
                rightBottomRadius: 10,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 150,
                  height: 150,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                rightTopRadius: 15,
                rightBottomRadius: 15,
                child: Container(
                  width: 300,
                  padding: const EdgeInsets.all(5),
                  color: Colors.blue,
                  child: Row(
                    children: <Widget>[
                      Expanded(
                          child: Column(
                        children: <Widget>[
                          Center(
                            child: Text("测试多种布局"),
                          ),
                          Text('支持不同角，不同圆弧大小')
                        ],
                      )),
                      Image.asset(
                        'assets/images/girl.jpeg',
                        fit: BoxFit.cover,
                        width: 60,
                        height: 60,
                      )
                    ],
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}


example/lib/main.dart
import 'package:flutter/material.dart';
import 'package:round_path_widget/round_path_widget.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        // This is the theme of your application.
        //
        // Try running your application with "flutter run". You'll see the
        // application has a blue toolbar. Then, without quitting the app, try
        // changing the primarySwatch below to Colors.green and then invoke
        // "hot reload" (press "r" in the console where you ran "flutter run",
        // or simply save your changes to "hot reload" in a Flutter IDE).
        // Notice that the counter didn't reset back to zero; the application
        // is not restarted.
        primarySwatch: Colors.blue,
      ),
      home: MyHomePage(),
    );
  }
}

class MyHomePage extends StatefulWidget {
  @override
  _MyHomePageState createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {


  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text("round_path_widget"),
      ),
      body: SingleChildScrollView(
        child: Center(
          child: Column(
            children: <Widget>[
              RoundPathWidget(
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 60,
                  height: 60,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.RoundRect,
                radius: 10,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 70,
                  height: 70,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                rightBottomRadius: 20,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 110,
                  height: 110,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                leftBottomRadius: 20,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 90,
                  height: 90,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                leftTopRadius: 20,
                leftBottomRadius: 10,
                rightBottomRadius: 10,
                child: Image.asset(
                  'assets/images/girl.jpeg',
                  fit: BoxFit.cover,
                  width: 150,
                  height: 150,
                ),
              ),
              RoundPathWidget(
                pathShape: PathShapeEnum.PartRoundRect,
                rightTopRadius: 15,
                rightBottomRadius: 15,
                child:Container(
                  padding: const EdgeInsets.all(3),
                  width: 300,
                  color: Colors.blue,
                  child:  Row(
                    children: <Widget>[
                      Expanded(child: Column(
                        children: <Widget>[
                          Center(child: Text("测试多种布局"),),
                          Text('支持不同角，不同圆弧大小')
                        ],
                      )),
                      Image.asset(
                        'assets/images/girl.jpeg',
                        fit: BoxFit.cover,
                        width: 60,
                        height: 60,
                      )
                    ],
                  ),
                ),
              ),
            ],
          ),
        ),
      ),
    );
  }
}
